plugin/Controller/EstateViewDocumentTitleBuilder.php

```diff
@@ -43,29 +43,20 @@ public function __construct(EstateTitleBuilder $pEstateTitleBuilder, WPQueryWrap
 		$this->_pEstateTitleBuilder = $pEstateTitleBuilder;
 		$this->_pWPQueryWrapper = $pWPQueryWrapper;
 	}
-
+	
 	/**
-	 * @param array $title see Wordpress internal function wp_get_document_title()
-	 * @return array
+	 * @param string $title see Wordpress internal function wp_get_document_title()
+	 * @param string $format
+	 * @return string
+	 * @throws Exception
 	 */
-	public function buildDocumentTitle(array $title): array
+	public function buildDocumentTitleField(string $format): string
 	{
 		$estateId = (int)$this->_pWPQueryWrapper->getWPQuery()->get('estate_id', 0);
 		if ($estateId === 0) {
-			return $title;
+			return '';
 		}
-
-		$title['title'] = $this->_pEstateTitleBuilder->buildTitle($estateId, '%1$s');
```

Comment:
With the current code, the plugin replaces the entrie `<title>` with just the estate's title, e.g. "Beautiful estate". It would be better if the title would still contain the title separator and site name like "Beautiful estate | My Real Estate".

I think one way of achieving this is using the [`document_title_parts` filter](https://developer.wordpress.org/reference/hooks/document_title_parts/), like this previous code did, and just replacing the `$title['title']` so that the other parts are kept.

Reply:
yes, i will check and fix it.

Reply:
@jmaas-onoffice  I checked the custom field for onoffice_id/onoffice_datensatznr work fine.Note You need add
![Screenshot](https://user-images.githubusercontent.com/104892108/174562968-8520bb93-36ed-4c3b-b4ac-1e52dc111746.png)
:

Reply:
> With the current code, the plugin replaces the entrie `<title>` with just the estate's title, e.g. "Beautiful estate". It would be better if the title would still contain the title separator and site name like "Beautiful estate | My Real Estate".
> 
> I think one way of achieving this is using the [`document_title_parts` filter](https://developer.wordpress.org/reference/hooks/document_title_parts/), like this previous code did, and just replacing the `$title['title']` so that the other parts are kept.

I think it is not necessary because All SEO plugins have variable for separator ex: Yoast "%%sep%%"

Reply:
@jmaas-onoffice What title format do you want to use?
"Title | Site Name" or like the old one with separator

![image](https://user-images.githubusercontent.com/101325858/174988959-90d48e0b-1e7b-479a-b5ad-173934c3c790.png)

Reply:
I think it's best to use [document_title_parts filter](https://developer.wordpress.org/reference/hooks/document_title_parts/) as described above, because then you only need to set the estate title and the rest is handled by WordPress.